great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -573,7 +574,8 @@ def get_batch_list_from_batch_request(
         splitter = self.splitter
         batch_spec_kwargs: dict[str, str | dict | None]
         for request in self._fully_specified_batch_requests(batch_request):
-            batch_metadata = copy.deepcopy(request.options)
+            batch_metadata = copy.deepcopy(self.batch_metadata)
```

Comment:
Since order matters here and I think we do this is a few places, I will put this in a helper method and replace all calls like this in my next PR.